src/indexer/segment_writer.rs

```diff
@@ -157,7 +158,11 @@ impl SegmentWriter {
 
     fn index_document(&mut self, doc: &Document) -> crate::Result<()> {
         let doc_id = self.max_doc;
-        for (field, values) in doc.get_sorted_field_values() {
+        let mut field_vals: Vec<_> = doc.field_values().iter().collect();
+        field_vals.sort_by_key(|el| el.field());
+        let iter = field_vals.into_iter().group_by(|el| el.field());
```

Comment:
`iter` is a sloppy name.

Can you call it `field_vals` for instance?

Reply:
yep, but I called it vals_grouped_by_field, since it's not an iter but IntoIter